presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1269,43 +1269,43 @@ private Context(
             this.nameToLambdaArgumentDeclarationMap = nameToLambdaArgumentDeclarationMap;
         }
 
-        public static Context notInLambda()
```

Comment:
I think the old way is correct as these methods are public factories for the class and thus are logically part of the class' public API. Visibility is controlled at the class level. Similarly, if we have a top level package-private classes, we'd make the methods public rather than package-private. In addition to being more clear about the public API, it's a one line change if the class later needs to become public (instead of changing / thinking about each method).

Reply:
Agreeable except this class is totally internal and won't ever be made public. I have reverted the changes though.